### PR TITLE
Fix buffergeometry merge function

### PR DESCRIPTION
**Pull request by wandergis**
Fix buffergeometry merge function, because typedarray's length can not be changed once it is initialized.

### CODE REVIEW

File: src/core/BufferGeometry.js [hunk 1]
```diff
@@ -816,14 +816,22 @@ Object.assign( BufferGeometry.prototype, EventDispatcher.prototype, {
...
-
+			var mergeArray = new attributeArray1.constructor(attributeArray1.length + attributeArray2.length);
+			mergeArray.set(attributeArray1);
+			mergeArray.set(attributeArray2, attributeArray1.length);
+			attributes[ key ] = new BufferAttribute(mergeArray, attribute1.itemSize);
 		}
 
+		// merge index
+		var index1 = this.index;
+		var indexArray1 = index1.array;
```
Comment (WestLangley) [inline src/core/BufferGeometry.js]: Not every buffer-geometry is of the indexed type.

**Comment by WestLangley** [comment]
@wandergis `BufferGeometry.merge()` currently only supports non-indexed buffer-geometries.

It also assumes the buffers are sufficiently sized to begin with.

**Comment by WestLangley** [comment]
@mrdoob I think `BufferGeometry.merge()` -- as currently implemented -- should be renamed to `BufferGeometry.copyAt()`.

So, to merge buffer-geometries, a pattern like this would make sense:
```javascript
var bufferGeometry = new THREE.BufferGeometry();

// allocate sufficiently-sized buffers
bufferGeometry.addAttribute( 'position', new THREE.Float32BufferAttribute( size * 3, 3 ) );
bufferGeometry.addAttribute( 'uv', new THREE.Float32BufferAttribute( size * 2, 2 ) );

// populate
bufferGeometry.copyAt( buf_geo_1, 0 );
bufferGeometry.copyAt( buf_geo_2, offset );
. . .
```

**Comment by mrdoob** [comment]
I think I would vote for removing `.merge()` in favor of #12804...

**Pull request closed without merging**